File: src/main/java/com/lessmarkup/dataobjects/Currency.java
package com.lessmarkup.dataobjects;

import com.lessmarkup.interfaces.data.AbstractDataObject;
import com.lessmarkup.interfaces.recordmodel.RequiredField;

import java.time.OffsetDateTime;

public class Currency extends AbstractDataObject {
    private String name;
    private String code;
    private double rate;
    private boolean enabled;
    private boolean isBase;
    private OffsetDateTime lastUpdate;

    @RequiredField
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }

    @RequiredField
    public String getCode() { return code; }
    public void setCode(String code) { this.code = code; }
    
    public double getRate() { return rate; }
    public void setRate(double rate) { this.rate = rate; }
    
    public boolean getEnabled() { return enabled; }
    public void setEnabled(boolean enabled) { this.enabled = enabled; }
    
    public boolean getIsBase() { return isBase; }
    public void setIsBase(boolean isBase) { this.isBase = isBase; }
    
    public OffsetDateTime getLastUpdate() { return lastUpdate; }
    public void setLastUpdate(OffsetDateTime lastUpdate) { this.lastUpdate = lastUpdate; }
}

File: src/main/scala/com/lessmarkup/dataobjects/Currency.scala
/*
 * This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
 * If a copy of the MPL was not distributed with this file, You can obtain one at
 * http://mozilla.org/MPL/2.0/.
 */

package com.lessmarkup.dataobjects

import com.lessmarkup.interfaces.annotations.RequiredField
import com.lessmarkup.interfaces.data.DataObject

class Currency extends DataObject {
  @RequiredField
  var name: String = null
  @RequiredField
  var code: String = null
  var rate: Double = 0
  var enabled: Boolean = false
  var isBase: Boolean = false
  var lastUpdate: Boolean = false
}